flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/AvgAggFunction.scala

```diff
@@ -321,12 +317,11 @@ class DecimalAvgAggFunction extends AggregateFunction[BigDecimal] {
     if (value != null) {
       val v = value.asInstanceOf[BigDecimal]
       val accum = accumulator.asInstanceOf[DecimalAvgAccumulator]
+      accum.f0 = accum.f0.subtract(v)
+      accum.f1 -= 1L
       if (accum.f1 == 0) {
```

Comment:
Do we need this check?
It only makes sense if we assume that accumulated and retracted values differ. But if that's the case all values for `cnt != 0` would be considered wrong.